### PR TITLE
Send HUP to dhcp6c

### PR DESCRIPTION
**Pull request by marjohn56**
Make use of updated dhcp6c clients config re-read. Also don't launch dhcp6c if it's already running.

### CODE REVIEW

File: src/etc/inc/interfaces.inc [hunk 1]
```diff
@@ -3117,11 +3117,14 @@ function interface_dhcpv6_prepare($interface = 'wan', $wancfg)
     $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_defaultgwv6\n";
     $rtsoldscript .= "fi\n";
     $rtsoldscript .= "if [ -f /var/run/dhcp6c_{$wanif}.pid ]; then\n";
-    $rtsoldscript .= "\t/bin/pkill -F /var/run/dhcp6c_{$wanif}.pid\n";
+    $rtsoldscript .= "\tdhcp6c_pid=\$(cat \"/var/run/dhcp6c_{$wanif}.pid\")\n";
+    $rtsoldscript .= "\t/bin/kill -HUP \${dhcp6c_pid}\n";
+    $rtsoldscript .= "\t/usr/bin/logger -t dhcpd \"RTSOLD script sending SIGHUP to dhcp6c\"\n";
     $rtsoldscript .= "\t/bin/sleep 1\n";
```
Comment (fichtner) [inline src/etc/inc/interfaces.inc]: can kill the sleep now :)

File: src/etc/inc/interfaces.inc [hunk 2]
```diff
@@ -3117,11 +3117,14 @@ function interface_dhcpv6_prepare($interface = 'wan', $wancfg)
     $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_defaultgwv6\n";
     $rtsoldscript .= "fi\n";
     $rtsoldscript .= "if [ -f /var/run/dhcp6c_{$wanif}.pid ]; then\n";
-    $rtsoldscript .= "\t/bin/pkill -F /var/run/dhcp6c_{$wanif}.pid\n";
+    $rtsoldscript .= "\tdhcp6c_pid=\$(cat \"/var/run/dhcp6c_{$wanif}.pid\")\n";
+    $rtsoldscript .= "\t/bin/kill -HUP \${dhcp6c_pid}\n";
```
Comment (fichtner) [inline src/etc/inc/interfaces.inc]: FWIW, there is such a thing:

    /bin/pkill -HUP -F /var/run/dhcp6c_{$wanif}.pid

**Comment by fichtner** [comment]
It just that: less code, less problems

**Comment by marjohn56** [comment]
Up to a point... Zero code = massive problems. :)

**Comment by marjohn56** [comment]
Now this is why the restart is useful now, this was just a save and apply to the WAN interface. This often caused multiple dhcp6 clients. Testing this looks good, I'll update the commit.

Mar 30 14:08:51 OPNsense dhcpd:  RTSOLD script - Starting dhcp6 client for interface wan(em0)
Mar 30 14:08:51 OPNsense dhcp6c[81822]: failed to open /usr/local/etc/dhcp6cctlkey: No such file or directory
Mar 30 14:08:51 OPNsense dhcp6c[81822]: failed initialize control message authentication
Mar 30 14:08:51 OPNsense dhcp6c[81822]: skip opening control port
Mar 30 14:08:52 OPNsense dhcp6c[82143]: Sending Solicit
Mar 30 14:08:52 OPNsense dhcp6c[82143]: restarting
Mar 30 14:08:52 OPNsense dhcpd: RTSOLD script sending SIGHUP to dhcp6c
Mar 30 14:08:52 OPNsense dhcp6c[82143]: Sending Solicit
Mar 30 14:08:53 OPNsense dhcp6c[82143]: Sending Request
Mar 30 14:08:53 OPNsense dhcp6c[82143]: dhcp6c Received REQUEST
Mar 30 14:08:53 OPNsense dhcp6c[82143]: add an address 2a02:8010:6228:2:20c:29ff:fe8b:e490/64 on em1
Mar 30 14:08:53 OPNsense dhcp6c[82143]: add an address 2a02:8010:6228:2:20c:29ff:fe8b:e49a/64 on em2
Mar 30 14:08:53 OPNsense dhcp6c[82143]: T1(2250) and/or T2(3600) is locally determined
Mar 30 14:08:53 OPNsense dhcp6c[82143]: add an address 2a02:8010:6228:0:eeee:c577:2816:844a/128 on em0
Mar 30 14:08:53 OPNsense dhcp6c[82143]: T1(2250) and/or T2(3600) is locally determined
Mar 30 14:08:53 OPNsense dhcp6c: dhcp6c REQUEST on em0 - running newipv6

**Comment by fichtner** [comment]
Merged, thank you.

**Comment by marjohn56** [comment]
And I think we'll leave it at that unless we hear something. Next thing to look at will be multiple WAN dhcp6c... but I'll leave that can of worms till after my holiday. :)

**Comment by fichtner** [comment]
@marjohn56 enjoy the holiday!

**Comment by marjohn56** [comment]
I will when I go, couple of weeks yet.

**Comment by fichtner** [comment]
Thought you meant Easter, but in both cases: enjoy :)

**Comment by fichtner** [comment]
@marjohn56 this works nicely at the office. very cool work, thanks again!

**Comment by marjohn56** [comment]
Excellent.. :)

**Comment by fichtner** [comment]
also, this: https://github.com/opnsense/dhcp6c

**Comment by marjohn56** [comment]
What about it? 😊

**Comment by fichtner** [comment]
It's our fork for easier work. Someone else asked for a new feature, too: https://twitter.com/noctarius2k/status/980881639764045824

**Pull request merged**